middleware/http/opa/middleware.go

```diff
@@ -43,6 +48,32 @@ var (
 	errOpaInvalidResultType = errors.New("got an invalid type back from repo policy. Only a boolean or map is valid")
 )
 
+func (s *Status) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		*s = Status(value)
+
+		return nil
+	case string:
+		intVal, err := strconv.Atoi(value)
+		if err != nil {
+			return err
+		}
+		*s = Status(intVal)
+
+		return nil
```

Comment:
Optional: you've done a lot of work on this already (thanks!), so this doesn't have to go in this PR, but it occurred to me that since `Status` is now a custom type, we can also enforce that it is in the correct range for RFC 2616 status codes (100-599), so instead of returning immediately from each case statement, add a range check after the switch-case before returning `nil` error.

Reply:
Done with tests.